### ISSUE TITLE
Trying to run simulation from generated network (using berlin example config files)

### ISSUE THREAD

**Issue by KreulitschD** [matsim-org/matsim-code-examples#175]
Hello,

I'm currently getting into MATSim and tried to run a simulation via the MATSim GUI of a small area of Berlin. I used JOSM and the MATSim Plugin as described in the slides from the TU homepage (https://isis.tu-berlin.de/pluginfile.php/981420/mod_resource/content/0/matstimjosm.pdf) to generate the network file and compressed it via 7-Zip (to get a xxx.xml.gz file). I thought I might be able to use the config file from the Berlin example but as it turns out the simulation doesn't work. (The Berlin example itself works just fine.)

- Is it possible to use the config file from Berlin or am I completely wrong?
- How could i change or generate a config file to get the simulation running?

Operating System: Windows 10
MATSim Version: 0.10.1 “Summer 2018”, released August 2018
Java Version: 1.8.0_211; Oracle Corporation; mixed mode; 64-bit
JOSM Version: 15155

Please find the logfile attached.
[logfile.log](https://github.com/matsim-org/matsim-code-examples/files/3377667/logfile.log)

Thank you very much in advance,
David

**Comment by nkuehnel** [comment]
Hi David,

from what I understand from your logfile it looks like the simulation crashed when preparing the plans for the simulation (which includes checking for routes). Could it be that your demand/plans file already includes routes from links that were created on another network?

> How could i change or generate a config file to get the simulation running?

besides editing the config in xml you can create a config in code like this:

`Config config = ConfigUtils.createConfig();`

or load an existing config:

`ConfigUtils.loadConfig(configFilename);`

Then, for each module that you find in the xml there is a respective config group in code that offers methods to adapt the settings, e.g. for the qsim:

`config.qsim().setNumberOfThreads(1);`

**Comment by kainagel** [comment]
The logfile points to a null pointer exception in a line that reads
```
		routeDistance += network.getLinks().get(networkRoute.getStartLinkId()).getLength() * (1.0 - relPosOnDepartureLink);
```
Presumably, `network.getLinks().get(...)` does not find the link with the given id, and then `null.getLength()` evidently can't work.

It looks like you are trying to use `plans_hwh_1pct.xml.gz` together with your own `network.xml.gz`.  This will not work, since the former is already registered to the network, i.e. it is referring to network links that do not exist in your network file.

What you need to do is to remove all references to links in `plans_hwh_1pct.xml.gz`.  This is easier said than done.  Note that this also implies removing the routes.  You could reduce the plans file to a single person and try with that.  However, you will most probably have more problems, such as missing transit routes, and then weird behavior by persons living far away from you network. 

We normally also keep plans files from before they are registered to the network, but it looks like these are not on the server.  I will try to find them and put them there.  But given that the person who has done this is on vacation, and afterwards with nearly no overlap I will be on vacation, this may take until september, sorry.

**Comment by KreulitschD** [comment]
Hello again,

thank you very much for your quick answer and sorry for my late response.
> from what I understand from your logfile it looks like the simulation crashed when preparing the plans for the simulation (which includes checking for routes). Could it be that your demand/plans file already includes routes from links that were created on another network?

That must be the case as i only generated the network myself and tried using already existing plans etc. from the Berlin example files. (Because i wasn't able to generate them myself.) So i would need to generate the needed config files for the choosen network as i expected.

> 
> > How could i change or generate a config file to get the simulation running?
> 
> besides editing the config in xml you can create a config in code like this:
> 
> `Config config = ConfigUtils.createConfig();`
> 
> or load an existing config:
> 
> `ConfigUtils.loadConfig(configFilename);`
> 
> Then, for each module that you find in the xml there is a respective config group in code that offers methods to adapt the settings, e.g. for the qsim:
> 
> `config.qsim().setNumberOfThreads(1);`

Thanks for your explanation but as i mentioned i'm completely new to MATSim (and JAVA) so i tried but couldn't figure out how to do that. I also tried editing the existing files with no positiv results.
Maybe there is a step by step guide/tutorial online (under the TU course MATSim Public Tutorial 2018) on how to generate the needed files for a self generated network? (Perhabs i have overlooked it.)

Basically i want to simulate and then edit and change different parameters to try various scenarios for the Mierendorffinsel in Berlin Charlottenburg.

Thank you very much in advance,
David

PS: I'm working at UdK Berlin as a research associate (on refurbishment strategies on district level) and was wondering if it would be possible to visit your department and get a closer look at the features of MATSim?

**Comment by kainagel** [comment]
> On 9. Sep 2019, at 16:25, KreulitschD <notifications@github.com> wrote:
> 
> PS: I'm working at UdK Berlin as a research associate (on refurbishment strategies on district level) and was wondering if it would be possible to visit your department and get a closer look at the features of MATSim?
> 
> 

Sure.  Either my "Sprechstunde", or make an appointment with Dominik Ziemke.  The coordinates are on the web under https://www.vsp.tu-berlin.de ; I hope you can find them; I don't really want to explicitly post them here so that the robots can find them even more easily.
